pkg/epp/handlers/response.go

```diff
@@ -64,13 +64,31 @@ func (s *StreamingServer) HandleResponseBody(ctx context.Context, reqCtx *Reques
 	return reqCtx, nil
 }
 
-// The function is to handle streaming response if the modelServer is streaming.
+// HandleResponseBodyModelStreaming processes a single chunk of a streaming response.
+// It accumulates token usage over the entire stream and records the final metrics only when the end-of-stream message
+// is detected.
 func (s *StreamingServer) HandleResponseBodyModelStreaming(ctx context.Context, reqCtx *RequestContext, responseText string) {
+	// Parse the current chunk for a 'usage' block.
+	resp := parseRespForUsage(ctx, responseText)
+
+	// Accumulate token counts.
```

Comment:
Two comments: 

1) From metrics perspective, we are not truly accumulating anything, this is just reporting the values in the last block.

2) Which logic depends `reqCtx.Usage` while we are accumulating the stream that would require us to keep updating the usage rather than just doing it at the end?

Reply:
I was trying to understand and debug behavior from integration test results, and I am not too familiar with our request stream handling code and properties we can rely upon in production. It's very possible my root cause analysis is wrong in places, but the symptoms I reported in the linked issues are reproducible.

Reply:
Great, thanks for confirming! I will fix the hermetic test setup to conform to this.

I think #1626 can be closed as WAI. I also reported an issue where metric labels were not being set properly, so I need to first confirm that this is solely triggered by the `data: [DONE]` final chunk case. If so, I'll revert my change to this response handling code.